### DISCUSSION TITLE
Call to Service Create not Broadcasting

### DISCUSSION THREAD

**Discussion by EricKulchycki** [feathersjs/feathers#2517] [Q&A] [answered]
Hello everyone, I'm new to feathers and I'm a little puzzled as to why my service calls aren't broadcasting. I could also be misunderstanding how it works.

As of right now I've got my server
```javascript
const app = express(feathers());
...
app.configure(socketio());
app.use("/readings", new ReadingService());
...
```
And I've got the reading service that has a `find` and `create` implemented (very basic for now, just trying to get it working).

On the client side, I've got a line of code that I would like to be listening for create events. Currently that is:
 (socket is app)
```javascript
const socket = io(url);
const app = feathers();

// Set up Socket.io client with the socket
app.configure(socketio(socket));
...
socket.service("readings").on("create", (payload) => {
  console.log({ message: "Readings Created", payload });
});
```

I've run a separate script that makes a client connection, and calls the create function in the service. I've verified that the function is called, but the above code in a separate client isn't being called. My understanding is that on the create an event should be broadcast, but there doesn't seem to be one. Any ideas on what I'm doing wrong?

Any help is appreciated, thanks!

**Comment by daffl** [comment, marked as answer]
Events will only be broadcast to connections that have been registered with a [channel](https://docs.feathersjs.com/api/channels.html) with the simplest example shown [in the API server quick start](https://docs.feathersjs.com/guides/basics/starting.html#an-api-server). In a generated application the standard behaviour is that events are only published to users that are authenticated. More details and links can be found in the `channels.js` file.

Additionally your client side code example probably won't work. It should be `app.service()` not `socket.service()` and `.on('created')` not `.on('create')`:

```js
const socket = io(url);
const app = feathers();

// Set up Socket.io client with the socket
app.configure(socketio(socket));
...
app.service("readings").on("created", (payload) => {
  console.log({ message: "Readings Created", payload });
});
```

I highly recomend following [the Feathers guide](https://docs.feathersjs.com/guides/) which discusses all those things.

**Reply by EricKulchycki** [reply]
Ahh thanks so much, I wasn't aware of the requirement of channels, and the need to declaratively publish events to certain channels.

As for the client code, app == socket, it's just a rename from a react hook, so that wasn't an issue, but created was right.

Thanks again!
